src/main/java/de/tum/cit/aet/artemis/programming/service/localci/LocalCIResultProcessingService.java

```diff
@@ -283,36 +280,34 @@ private BuildJob saveFinishedBuildJob(BuildJobQueueItem queueItem, BuildStatus b
         }
     }
 
-    private void updateExerciseBuildDurationAsync(ProgrammingExercise exercise) {
-        CompletableFuture.runAsync(() -> updateExerciseBuildDuration(exercise));
+    private void updateExerciseBuildDurationAsync(long exerciseId) {
+        CompletableFuture.runAsync(() -> updateExerciseBuildDuration(exerciseId));
     }
 
-    private void updateExerciseBuildDuration(ProgrammingExercise exercise) {
+    private void updateExerciseBuildDuration(long exerciseId) {
         try {
-            ProgrammingExerciseBuildStatistics buildStatistics = programmingExerciseBuildStatisticsRepository.findByExerciseId(exercise.getId()).orElse(null);
-            long successfulBuildJobCountByExerciseId = buildJobRepository.fetchSuccessfulBuildJobCountByExerciseId(exercise.getId());
-
-            boolean hasSuccessfulBuildJobs = successfulBuildJobCountByExerciseId > 0;
-            boolean exceedsUpdateThreshold = buildStatistics == null
-                    || successfulBuildJobCountByExerciseId - buildStatistics.getBuildCountWhenUpdated() >= BUILD_STATISTICS_UPDATE_THRESHOLD;
-
-            boolean shouldUpdate = hasSuccessfulBuildJobs && exceedsUpdateThreshold;
-            if (!shouldUpdate) {
+            var buildStatisticsDto = buildJobRepository.findBuildJobStatisticsByExerciseId(exerciseId);
+            if (buildStatisticsDto == null || buildStatisticsDto.buildCountWhenUpdated() == 0) {
                 return;
             }
+            var programmingExerciseBuildStatistics = programmingExerciseBuildStatisticsRepository.findByExerciseId(exerciseId).orElse(null);
 
-            OptionalDouble averageBuildDuration = buildJobRepository.fetchSuccessfulBuildJobsByExerciseIdWithLimit(exercise.getId(), BUILD_JOB_DURATION_UPDATE_LIMIT).stream()
-                    .mapToLong(buildJob -> Duration.between(buildJob.getBuildStartDate(), buildJob.getBuildCompletionDate()).toSeconds()).average();
-            if (averageBuildDuration.isPresent()) {
-                if (buildStatistics == null) {
-                    buildStatistics = new ProgrammingExerciseBuildStatistics(exercise.getId(), Math.round(averageBuildDuration.getAsDouble()), successfulBuildJobCountByExerciseId);
-                }
-                else {
-                    buildStatistics.setBuildDurationSeconds((long) averageBuildDuration.getAsDouble());
-                    buildStatistics.setBuildCountWhenUpdated(successfulBuildJobCountByExerciseId);
+            long averageDuration = Math.round(buildStatisticsDto.buildDurationSeconds());
+
+            if (programmingExerciseBuildStatistics == null) {
```

Comment:
Imo it's a bit weird that you first try to extract the value from the optional, to then use the manually set null to compare. 
I think it would be cleaner if you'd not ose the `.orElse(null)` here, and then simply use the `isEmpty()` in the condition.